v2/internal/controllers/to_azure_secrets_test.go

```diff
@@ -66,10 +71,68 @@ func Test_MissingSecretKey_ReturnsError(t *testing.T) {
 
 	tc.CreateResourceAndWaitForState(vm, metav1.ConditionFalse, conditions.ConditionSeverityWarning)
 	// We expect the ready condition to include details of the error
-	tc.Expect(vm.Status.Conditions[0].Reason).To(Equal("SecretNotFound"))
+	tc.Expect(vm.Status.Conditions[0].Reason).To(Equal(conditions.ReasonSecretNotFound))
 	tc.Expect(vm.Status.Conditions[0].Message).To(
 		ContainSubstring("Secret \"%s/%s\" does not contain key \"%s\"", tc.Namespace, secret.Name, secret.Key))
 
 	// Delete VM and resources.
 	tc.DeleteResourcesAndWait(vm, networkInterface, subnet, vnet, rg)
 }
+
+func Test_UserSecretInDifferentNamespace_SecretNotFound(t *testing.T) {
```

Comment:
I don't think that this test will detect #2398 (though it's a fine test to have).

To detect the issue with #2398, what we need is 2 of the same kind of resource, each in different namespaces, where each is looking at a secret with the same name. Something like:

1. Create vm1 in ns1, with secret named "mysecret" also in ns1. Allow this to be created fully and get to successfully provisioned.
2. Create vm2 in ns2, with secret ALSO named "mysecret" but in ns2. Allow this to be created fully and to get successfully provisioned.
3. Ensure that step 2 doesn't trigger a reconcile of vm1.

Item 3 is the "hard part" because we don't have great visibility into what reconciliations have happened in the tests, and it's hard to get that visibility in a way that will work in both envtest and real k8s clusters. You can easily see from logs, but that's not that helpful. Because nothing has changed you won't be able to tell by looking at the Ready condition of vm1.

The best idea I have to deal with item 3 is: use the azure client to go behind the back and delete vm1 before creating vm2. The operator won't realize it's been deleted. Then create vm2. Then confirm vm1 is still deleted in Azure. If vm1 was recreated in Azure, that means that reconcile was triggered which is unexpected: test fails.